cmake/EnableCompilerWarnings.cmake

```diff
@@ -67,7 +67,6 @@ else()
             -Wunused
             -Wno-ignored-qualifiers
             -Wno-sign-compare
-            -Wno-deprecated
```

Comment:
@averinevg why this change? I recall we added it due to Ubuntu 22.04 and Clang-16. Has the issue been fixed?

Reply:
@junliume I need this to be able to mark deprecated functions. I've reverted it back here and created https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1942. Thanks!